src/typescript/frontend/src/app/chats/route.ts

```diff
@@ -0,0 +1,43 @@
+import { fetchChatEvents } from "@/queries/market";
+import type { NextRequest } from "next/server";
+import { stringifyJSON } from "utils";
+
+type ChatSearchParams = {
+  marketID: string | null;
+  fromMarketNonce: string | null;
+};
+
+export type ValidChatSearchParams = {
+  marketID: string;
+  fromMarketNonce: string;
+};
+
+const isNumber = (s: string) => !isNaN(parseInt(s));
```

Comment:
```suggestion
const isNumber = (s: string) => {
  try {
    return !isNaN(parseInt(s));
  } catch {
    return false;
  }
}
```

Should always wrap these with a try/catch since it's coming from a search param and a malformed request somewhere will throw an error

Reply:
I'll note that `parseInt("lol")` just returns `NaN` so I don't think this is a problem.